contract_compassion/view/contract_view.xml

```diff
@@ -0,0 +1,172 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+	<data>
+		<record id="view_recurring_contract_tree_compassion" model="ir.ui.view">
+			<field name="name">recurring.contract.compassion.tree</field>
+			<field name="model">recurring.contract</field>
+			<field name="inherit_id" ref="recurring_contract.view_recurring_contract_tree"/>
+			<field eval="18" name="priority"/>
+			<field name="arch" type="xml">
+				<data>
+					<tree string="Contracts" position="attributes">
+						<attribute name="colors">blue:state == 'draft';red:state == 'mandate';green:state == 'waiting';black:state == 'active';gray:state in ('terminated','cancelled')</attribute>
+						<attribute name="string">Sponsorships</attribute>
+					</tree>
+					<field name="partner_id" position="after">
+						<field name="child_code" />
+						<field name="child_name" />
+					</field>
+					<field name="partner_id" position="before">
+						<field name="partner_codega" invisible="context.get('partner_contracts_view')" />
+					</field>
+                    <field name="payment_term_id" position="after">
+                        <field name="frequency" />
+                        <field name="origin_id" />
+                    </field>
+                    <field name="state" position="after">
+                        <button name="open_contract" type="object" invisible="not context.get('partner_contracts_view')" icon="STOCK_JUMP_TO" />
+                    </field>
+					<field name="reference" position="replace" />
+				</data>
+			</field>
+		</record>
+		
+		<record id="view_recurring_contract_form_compassion" model="ir.ui.view">
+			<field name="name">recurring.contract.compassion.form</field>
+			<field name="model">recurring.contract</field>
+			<field name="inherit_id" ref="recurring_contract.view_recurring_contract_form"/>
+            <field eval="14" name="priority"/>
+            <field name="mode">primary</field>
+			<field name="arch" type="xml">
+                <!-- Modify header to add the waiting state -->
+                <field name="state" position="attributes">
+                    <attribute name="statusbar_visible">draft,active,waiting,terminated</attribute>
+                </field>
+
+                <button name="contract_terminated" position="attributes">
+                    <attribute name="states" position="replace"/>
+                    <attribute name="attrs">{'invisible':['|',('state', '!=', 'active'),('type', '!=', 'O')]}</attribute>

```

Comment:
Because we need to have a button that launch a wizard to end a sponsorship contract and another that launch a simple confirm popup.